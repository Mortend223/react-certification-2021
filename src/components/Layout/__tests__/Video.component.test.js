import React from 'react';
import { render, screen } from '@testing-library/react'


// Components
import VideoComponent from '../Video.component';

const props = {
    description: 'some text', title: 'title of video', url: '/logo512.png'
}

describe('Video Component', () => {
    it('Should show Video Component', () => {
      const container = render(<VideoComponent {...props}/>)

      expect(container.firstChild).toMatchSnapshot()
    });
    it('Should render the title', () => {
        render(<VideoComponent {...props}/>)

        expect(screen.getByText(props.title)).toBeInTheDocument()
    });
    it('Should render the thumbnail', () => {
        render(<VideoComponent {...props}/>)
        const thumbnail = screen.getByRole('img');

        expect(thumbnail).toHaveAttribute('src','/logo512.png');
    });
    it('Should render the description', () => {
        render(<VideoComponent {...props}/>)

        expect(screen.getByText(props.description)).toBeInTheDocument()
    });
});