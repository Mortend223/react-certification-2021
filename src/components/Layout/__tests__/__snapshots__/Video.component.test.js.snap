// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video Component Should show Video Component 1`] = `undefined`;
